builtin/credential/aws/backend.go

```diff
@@ -174,6 +181,9 @@ func (b *backend) periodicFunc(req *logical.Request) error {
 			b.tidyWhitelistIdentity(req.Storage, safety_buffer)
 		}
 
+		// get rid of old unique ID entries
+		b.cleanOldCachedUniqueIdMapping()
```

Comment:
Do you see a reason to do it like this instead of using go-cache, since this is purely in-memory? I don't have a strong preference, go-cache will just do it internally and individually rather than on a schedule in a block.

Reply:
This would also get rid of the need to hold a mutex to do this operation and the need to store a last-accessed time. The downside, mainly, is that updating the expiration time still requires a replace operation (but the locking is handled internally).

Reply:
Because I wasn't aware go-cache existed :) I was mainly pattern matching off of the cached AWS clients. And hopefully patrickmn/go-cache#66 will get merged to make it easier to update the expiration time.

I'll work on moving to go-cache and push a new commit when ready.

Reply:
Done!